redash/handlers/embed.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from flask import render_template, request, redirect, session, url_for, flash
+from flask.ext.restful import abort
+
+from flask_login import current_user, login_required
+
+from redash import models, settings
+from redash.wsgi import app
+from redash.utils import json_dumps
+
+@app.route('/embed/query/<query_id>/visualization/<visualization_id>', methods=['GET'])
+@login_required
+def embed(query_id, visualization_id):
+
+    v = models.Visualization.get(models.Visualization.id == visualization_id)
```

Comment:
Another alternative is: `vis = models.Visualization.get(models.Visualization.id == ..., models.Visualization.query == ...)` - this will throw `models.Visualization.DoesNotExist` if not found (you can do `select()` with `first()` instead, which won't throw iirc).